### DISCUSSION TITLE
Flows: What exactly happens with `response` data in a blocking event hook?

### DISCUSSION THREAD

**Discussion by erondpowell** [directus/directus#15601] [Archive]
**How is the `response` data used on the blocking event trigger?**

**When we create a trigger with a "blocking" event hook, we have the option to set a `response`.**

![Screen Shot 2022-09-15 at 1 25 08 PM](https://user-images.githubusercontent.com/62306630/190481256-df1b66fa-34e1-43b9-87cd-fb16fbc731f6.png)

**This `response` data is added onto the Flow object under a `return` key...**

![Screen Shot 2022-09-15 at 1 27 31 PM](https://user-images.githubusercontent.com/62306630/190481645-b28c34cc-47c6-42dd-968c-06b006be5aa2.png)

**I was assuming `response` data made the Flow add the `response` data into the *create item post request*, but it appears that's not the case.**

![CleanShot 2022-09-15 at 13 31 30@2x](https://user-images.githubusercontent.com/62306630/190482411-4e03a7a5-d661-4e25-a2f9-4536d16783ac.png)

---

**Additionally, if I choose the "last operation" or "all data":**

![Screen Shot 2022-09-15 at 2 04 09 PM](https://user-images.githubusercontent.com/62306630/190488058-a9cbd71b-d251-4f00-a290-e7b8f17b5bee.png)

**When I try to create an Item, the item is created with a `NULL` field value(s).**

![CleanShot 2022-09-15 at 14 01 32@2x](https://user-images.githubusercontent.com/62306630/190487613-348cffb1-8480-4cad-911b-365185181d3c.png)

**Why does this happen? Is this a bug? or misconfiguration?**
---
**Thanks team!** 🙏

**Comment by licitdev** [comment]
It looks like a misconfiguration. 👀 

The `Response Body` defines the return value after the Flow completes.
The first two options represent `$last` and `$all` respectively.
The self typed other value should be the key of an operation.

Your flow can be `trigger->1->2->3->4`, but the updated payload can be what `2` returns.

Here's a sample flow of an item creation event.

1. Server receives the POST request with the payload `{ "title": "hello docs 2.0" }`.
2. Server triggers the `dummy_articles.items.create` filter event with the payload.
3. The operations within the flow gets triggered accordingly.
4. If any operation errors out or returns thru the reject route, the item creation is aborted.
5. If all operation succeeds, it will return a value based on the `Response Body` setting.

Essentially, with the filter event (blocking), one is able to perform validation and manipulation on the payload. The returned value will be used when creating the item.

### Reference 

https://github.com/directus/directus/blob/f9c4422fb177ac07a2f83cbeaa808f3c9e3e02a2/api/src/services/items.ts#L99-L117
